icicle/tests/test_curve_api.cpp

```diff
@@ -236,57 +218,55 @@ TEST_F(CurveApiTest, ecntt_new)
+    ICICLE_CHECK(icicle_malloc((void**)&d_out, total_size * sizeof(projective_t)));
+    ICICLE_CHECK(icicle_copy(d_in, input.get(), total_size * sizeof(projective_t)));
+
     auto config = default_ntt_config<scalar_t>();
-    config.stream = stream;
     config.batch_size = batch_size;       // default: 1
     config.columns_batch = columns_batch; // default: false
     config.ordering = ordering;           // default: kNN
     config.are_inputs_on_device = true;
     config.are_outputs_on_device = true;
-    config.is_async = false;
-    ICICLE_CHECK(ntt_init_domain(scalar_t::omega(log_ntt_domain_size), init_domain_config));
-    projective_t *d_in, *d_out;
-    ICICLE_CHECK(icicle_malloc_async((void**)&d_in, total_size * sizeof(projective_t), config.stream));
-    ICICLE_CHECK(icicle_malloc_async((void**)&d_out, total_size * sizeof(projective_t), config.stream));
-    ICICLE_CHECK(icicle_copy_to_device_async(d_in, input.get(), total_size * sizeof(projective_t), config.stream));
+
     std::ostringstream oss;
     oss << dev_type << " " << msg;
     START_TIMER(NTT_sync)
     for (int i = 0; i < iters; ++i) {
-      if (inplace) {
-        ICICLE_CHECK(ntt(d_in, N, dir, config, d_in));
-      } else {
-        ICICLE_CHECK(ntt(d_in, N, dir, config, d_out));
-      }
+      ICICLE_CHECK(ntt(d_in, N, dir, config, inplace ? d_in : d_out));
     }
     END_TIMER(NTT_sync, oss.str().c_str(), measure);
 
-    if (inplace) {
-      ICICLE_CHECK(icicle_copy_to_host_async(out, d_in, total_size * sizeof(projective_t), config.stream));
-    } else {
-      ICICLE_CHECK(icicle_copy_to_host_async(out, d_out, total_size * sizeof(projective_t), config.stream));
-    }
-    ICICLE_CHECK(icicle_free_async(d_in, config.stream));
-    ICICLE_CHECK(icicle_free_async(d_out, config.stream));
-    ICICLE_CHECK(icicle_stream_synchronize(config.stream));
-    ICICLE_CHECK(icicle_destroy_stream(stream));
+    ICICLE_CHECK(
+      icicle_copy_to_host_async(out, inplace ? d_in : d_out, total_size * sizeof(projective_t), config.stream));
+
+    ICICLE_CHECK(icicle_free(d_in));
+    ICICLE_CHECK(icicle_free(d_out));
+
     ICICLE_CHECK(ntt_release_domain<scalar_t>());
   };
-  // run(s_main_target, out_main.get(), "ecntt", false /*=measure*/, 10 /*=iters*/); // warmup
-  // run(s_ref_target, out_ref.get(), "ecntt", VERBOSE /*=measure*/, 10 /*=iters*/);
-  // run(s_main_target, out_main.get(), "ecntt", VERBOSE /*=measure*/, 10 /*=iters*/);
-  // ASSERT_EQ(0, memcmp(out_main.get(), out_ref.get(), total_size * sizeof(scalar_t))); // FIXME
+
+  run(s_main_target, out_main.get(), "ecntt", false /*=measure*/, 1 /*=iters*/); // warmup
+  run(s_ref_target, out_ref.get(), "ecntt", VERBOSE /*=measure*/, 1 /*=iters*/);
+  run(s_main_target, out_main.get(), "ecntt", VERBOSE /*=measure*/, 1 /*=iters*/);
+  // note that memcmp is tricky here because projetive points can have many representations
+  for (uint64_t i = 0; i < N; ++i) {
+    ASSERT_FALSE(projective_t::is_zero(out_ref[i]));
```

Comment:
Is there a chance for a valid calculation to result in zero?

Reply:
Input is 0 so if we do nothing inside the backend the tests passes otherwise so I want to avoid this case.